core/domain/stats_domain.py

```diff
@@ -16,213 +16,250 @@
-        self.answers_list = answers_list
+        self.submitted_answer_list = submitted_answer_list
+        self.schema_version = schema_version
+
+    def get_submitted_answer_dict_list(self):
+        """Returns the submitted_answer_list stored within this object as a list
+        of StateAnswer dicts.
+        """
+        return [state_answer.to_dict()
+                for state_answer in self.submitted_answer_list]
 
     def validate(self):
         """Validates StateAnswers domain object entity."""
 
         if not isinstance(self.exploration_id, basestring):
             raise utils.ValidationError(
-                'Expected exploration_id to be a string, received %s' %
-                self.exploration_id)
+                'Expected exploration_id to be a string, received %s' % str(
+                    self.exploration_id))
 
         if not isinstance(self.state_name, basestring):
             raise utils.ValidationError(
-                'Expected state_name to be a string, received %s' %
-                self.state_name)
+                'Expected state_name to be a string, received %s' % str(
+                    self.state_name))
 
         if self.interaction_id is not None:
             if not isinstance(self.interaction_id, basestring):
                 raise utils.ValidationError(
-                    'Expected interaction_id to be a string, received %s' %
-                    str(self.interaction_id))
+                    'Expected interaction_id to be a string, received %s' % str(
+                        self.interaction_id))
 
-            # check if interaction_id is valid
+            # Verify interaction_id is valid.
             if (self.interaction_id not in
                     interaction_registry.Registry.get_all_interaction_ids()):
                 raise utils.ValidationError(
-                    'Unknown interaction id %s' % self.interaction_id)
+                    'Unknown interaction_id: %s' % self.interaction_id)
+
+        if not isinstance(self.submitted_answer_list, list):
+            raise utils.ValidationError(
+                'Expected submitted_answer_list to be a list, received %s' %
+                str(self.submitted_answer_list))
+
+        if not isinstance(self.schema_version, int):
+            raise utils.ValidationError(
+                'Expected schema_version to be an integer, received %s' % str(
+                    self.schema_version))
+
+        if self.schema_version < 1:
+            raise utils.ValidationError(
+                'schema_version < 1: %d' % self.schema_version)
+
+        if self.schema_version > feconf.CURRENT_STATE_ANSWERS_SCHEMA_VERSION:
+            raise utils.ValidationError(
+                'schema_version > feconf.CURRENT_STATE_ANSWERS_SCHEMA_VERSION '
+                '(%d): %d' % (
+                    feconf.CURRENT_STATE_ANSWERS_SCHEMA_VERSION,
+                    self.schema_version))
+
+        for submitted_answer in self.submitted_answer_list:
```

Comment:
That's actually what happens now (this validate is only called on a subset of answers being inserted), but I moved it out of here for consistency with other validate() methods.